easypy/timing.py

```diff
@@ -1,11 +1,97 @@
 from contextlib import contextmanager
 import time
+from math import inf
 from datetime import datetime, timedelta
 
 from .decorations import parametrizeable_decorator
 from .units import Duration
 
 
+def to_timestamp(t):
+    return t.timestamp() if isinstance(t, datetime) else t
+
+
+class TimeInterval(object):
+    __slots__ = ("t0", "t1")
+
+    NO_START = -inf
+    NO_END = inf
+
+    def __init__(self, from_time=NO_START, to_time=NO_END):
+        self.t0 = self.NO_START if from_time is None else to_timestamp(from_time)
+        self.t1 = self.NO_END if to_time is None else to_timestamp(to_time)
+
+    @property
```

Comment:
Why not `@safe_property`?

Reply:
we don't use it anywhere else in `easypy`, so I don't see a reason to start specifically in this commit.

I think property functions should be trivial enough not to require this added complication and dependency.